web/portal/user/src/router/useEntityMap.ts

```diff
@@ -1,9 +1,22 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 
+import { useStoreState } from '../store';
 import getEntityMap, { ExtendedRouteMap } from './EntityMap';
+import useFeatureFilteredEntityMap from './useFeatureFilteredEntityMap';
 
 export default function useEntityMap(): ExtendedRouteMap {
-  const [entityMap] = useState<ExtendedRouteMap>(getEntityMap());
+  const status = useStoreState((state) => state.userStatus.status.profile);
 
-  return entityMap;
+  const [entityMap, setEntityMap] = useState<ExtendedRouteMap>(getEntityMap());
+
+  useEffect(() => {
+    setEntityMap(getEntityMap());
+  }, [status]);
+
+  const aclFilteredEntityMap = useFeatureFilteredEntityMap({
+    entityMap,
+    status,
+  });
```

Comment:
We could optionally optimize by calculating entityMap just once. Something like
```suggestion
  const [entityMap, setEntityMap] = useState<ExtendedRouteMap>([]);

  useEffect(() => {
    if (entityMap.length > 0) {
      return;
    }
    setEntityMap(getEntityMap());
  }, [status]);

  const aclFilteredEntityMap = useFeatureFilteredEntityMap({
    entityMap,
    status,
  });
```

Reply:
I have done it but the component still gets reloaded.